src/types.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2016 The Incremental DOM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/** @typedef {string|function(new: Element): Element} */
```

Comment:
Going to just punt on this for now and take `!Function`. That allows people to use a workaround like the one in Amp, or just use plain ES6 classes if not transpiling. Lose a bit of type-safety but not a huge deal.